azurerm/internal/services/keyvault/key_vault_access_policy_resource.go

```diff
@@ -203,6 +205,28 @@ func resourceArmKeyVaultAccessPolicyCreateOrDelete(d *schema.ResourceData, meta
 	if _, err = client.UpdateAccessPolicy(ctx, resourceGroup, vaultName, action, parameters); err != nil {
 		return fmt.Errorf("Error updating Access Policy (Object ID %q / Application ID %q) for Key Vault %q (Resource Group %q): %+v", objectId, applicationIdRaw, vaultName, resourceGroup, err)
 	}
+	stateConf := &resource.StateChangeConf{
+		Pending:                   []string{"notfound", "vaultnotfound"},
+		Target:                    []string{"found"},
+		Refresh:                   accessPolicyRefreshFunc(ctx, client, resourceGroup, vaultName, objectId, applicationIdRaw),
+		Delay:                     5 * time.Second,
+		ContinuousTargetOccurence: 3,
+		Timeout:                   d.Timeout(schema.TimeoutCreate),
+	}
+
+	if action == keyvault.Remove {
+		stateConf.Target = []string{"notfound"}
+		stateConf.Pending = []string{"found", "vaultnotfound"}
+		stateConf.Timeout = d.Timeout(schema.TimeoutDelete)
+	}
+
+	if action == keyvault.Replace {
+		stateConf.Timeout = d.Timeout(schema.TimeoutUpdate)
+	}
+
+	if _, err := stateConf.WaitForState(); err != nil {
+		return fmt.Errorf("failed waiting for Key Vault Access Policy (Object ID: %q) to apply: %+v", objectId, err)
+	}
```

Comment:
(as discussed offline) this likely wants adding to the key vault resource's update function, too - to ensure that's covered from this edge-case too?